.github/actions/setup-rust/action.yml

```diff
@@ -0,0 +1,38 @@
+name: "Setup Rust"
+description: "Sets up the correct Rust version and caching via sccache and a GCP backend"
+inputs:
+  targets:
+    description: "Additional targets to install"
+    required: false
+    default: ""
+runs:
+  using: "composite"
+  steps:
+    - id: auth
+      uses: google-github-actions/auth@v1
+      with:
+        workload_identity_provider: "projects/397012414171/locations/global/workloadIdentityPools/github-actions-pool/providers/github-actions"
+        service_account: "github-actions@github-iam-387915.iam.gserviceaccount.com"
+    - run: |
+        echo "SCCACHE_GCS_BUCKET=firezone-sccache" >> $GITHUB_ENV
```

Comment:
👍

Reply:
> We should create this bucket using Terraform @thomaseizinger

Yeah that makes sense. I didn't think of it to be honest but testing that would have been a pain because it doesn't get applied until merge right?